src/backend/access/transam/xact.c

```diff
@@ -2317,14 +2317,14 @@ CommitTransaction(void)
 	willHaveObjectsFromSmgr =
 			PersistentEndXactRec_WillHaveObjectsFromSmgr(EndXactRecKind_Commit);
 
-	if (willHaveObjectsFromSmgr)
-	{
-		/*
-		 * We need to ensure the recording of the [distributed-]commit record and the
-		 * persistent post-commit work will be done either before or after a checkpoint.
-		 */
-		CHECKPOINT_START_LOCK;
-	}
+	/* In previous version, we ensured the recording of the [distributed-]commit record and the
+	 * persistent post-commit work will be done either before or after a checkpoint.
+	 *
+	 * However the persistent table status will be synchronized with AOSeg_XXXX
+	 * table and hdfs file in PersistentRecovery_Scan() at recovery PASS2.
+	 * We don't need to worry about inconsistent states between them. So no
+	 * CHECKPOINT_START_LOCK any more.
+	 */
```

Comment:
Now hawq doesn't report error when (4) occurs: I set break in gdb at xact.c:2427, and run below statements, when hang at the "COMMIT", then I run "hadoop dfsadmin -safemode enter" to set hdfs to save mode, then continue in gdb, the commit successfully finished with only warning. Similar problem for ABORT of transaction which includes 'CREATE TABLE'. 

postgres=# begin transaction ISOLATION LEVEL SERIALIZABLE;BEGIN
postgres=# drop table tableinfs2;                                                                                                                             DROP TABLE
postgres=# commit;
WARNING:  could not remove relation directory 24974/16387/24975: Input/output error
CONTEXT:  Dropping file-system object -- Relation Directory: '24974/16387/24975'
COMMIT
postgres=# select * from tableinfs2;
ERROR:  relation "tableinfs2" does not exist

As for your question above, the recovery process will redo (4) again and similarly it report warning info if we failed to drop file and or fail to modify persistent table. Thanks.